src/csharp/Microsoft.Spark/Utils/UdfSerDe.cs

```diff
@@ -68,7 +71,50 @@ internal sealed class FieldData
         {
             public TypeData TypeData { get; set; }
             public string Name { get; set; }
+            public ValueData ValueData { get; set; }
+        }
+
+        [Serializable]
+        internal sealed class ValueData : ISerializable
+        {
+            public ValueData() { }
+
+            public TypeData TypeData { get; set; }
+
             public object Value { get; set; }
+
+            public void GetObjectData(SerializationInfo info, StreamingContext context)
+            {
+                info.AddValue("TypeData", TypeData, typeof(TypeData));
+
+                var bf = new BinaryFormatter();
+                using (var ms = new MemoryStream())
+                {
+                    bf.Serialize(ms, Value);
```

Comment:
I guess null case is also tested?

Reply:
null case is currently checked in
```csharp
        private static TargetData SerializeTarget(object target)
        {
...
                    ValueData = new ValueData()
                    {
                        TypeData = (value != null) ? SerializeType(value.GetType()) : default,
                        Value = value
                    }
```

Reply:
But this is a serializer for `ValueData`. What will happen if `Value` is set to null? `bf.Serialize(ms, Value);` is still safe?

Reply:
Modified so ValueData can only be created via constructor.  Will throw an exception if value is null.  This will prevent TypeData and Value properties getting set to null.